middle_end/flambda2/naming/bindable_let_bound.ml

```diff
@@ -229,7 +229,12 @@ let must_be_symbols t =
   match t with
   | Symbols symbols -> symbols
   | Singleton _ | Set_of_closures _ ->
-    Misc.fatal_errorf "Bound name is not a [Set_of_closures]:@ %a" print t
+    Misc.fatal_errorf "Bound name is not a [Symbols]:@ %a" print t
+
+let must_be_symbols_opt t =
```

Comment:
I'd suggest either renaming the function to `may_be_symbols`, to show that the non-symbol case is expected, or (if you feel up to a bit of refactoring) to write a function `fold_all_bound_names` with the following signature:
```ocaml
val fold_all_bound_names
   : t
  -> init:'a
  -> var:('a -> Var_in_binding_pos.t -> 'a)
  -> symbol:('a -> Symbol.t -> 'a)
  -> code_id:('a -> Code_id.t -> 'a)
  -> 'a
```
Then you can call it directly in `Closure_conversion_aux.Let_with_acc.create`.